python/src/aiconfig/editor/client/src/components/prompt/model_settings/ModelSettingsRenderer.tsx

```diff
@@ -27,23 +29,40 @@ export default memo(function ModelSettingsRenderer({
   onUpdateModelSettings,
 }: Props) {
   const { classes } = useStyles();
-  let settingsComponent;
-
-  if (schema) {
-    settingsComponent = (
-      <ModelSettingsSchemaRenderer
-        settings={settings}
-        schema={schema}
-        onUpdateModelSettings={onUpdateModelSettings}
-      />
-    );
-  } else if (settings) {
-    settingsComponent = <ModelSettingsConfigRenderer settings={settings} />;
-  }
+  const [isRawJSON, setIsRawJSON] = useState(schema == null);
 
   return (
     <Flex direction="column" className={classes.settingsContainer}>
-      {settingsComponent}
+      {/* // TODO: Refactor this out to a generic wrapper for toggling JSONRenderer or children component */}
+      {/* // Only show the toggle if there is a schema to toggle between JSON and custom schema renderer */}
+      {schema && (
+        <Flex justify="flex-end">
+          <Tooltip label="Toggle JSON editor" withArrow>
+            <ActionIcon onClick={() => setIsRawJSON((curr) => !curr)}>
+              {isRawJSON ? (
+                <IconBracesOff size="1rem" />
+              ) : (
+                <IconBraces size="1rem" />
+              )}
+            </ActionIcon>
+          </Tooltip>
+        </Flex>
+      )}
+      {isRawJSON || !schema ? (
+        <JSONRenderer
+          content={settings}
+          onChange={(val) =>
+            onUpdateModelSettings(val as Record<string, unknown>)
+          }
+          // schema={schema} TODO: Add schema after fixing z-index issue
```

Comment:
What's z-index issue?

Reply:
<img width="578" alt="Screenshot 2024-01-05 at 10 48 37 AM" src="https://github.com/lastmile-ai/aiconfig/assets/5060851/ad229000-398b-4920-8e89-d20bdcad4b7e">

The editor tooltips have lower z-index than the surrounding elements so they're hidden. Just need to look into it and bump it